### PR TITLE
controlled exception searching Holder, and moved to helper

### PR DESCRIPTION
**Pull request by raul-fuentes**
Where
=====
* **Related Issue:** https://github.com/AyuntamientoMadrid/agendas/issues/311

What
====
- Controlled error searching Holder, and moved to a helper

### CODE REVIEW

**Comment by Senen** [comment]
Hi @raul-fuentes, i think we can have a better approach to solve this situation.

The problem occurs when a request for a non existent holder arrives.

Maybe we can solve this at controller level if we try to find holder with given params[:holder], if we cant find it we should redirect to /visitors with alert notice: "El titular solicitado no existe"

```
  def agenda
    @holder = Holder.where(id: params[:holder]).first
    if @holder.blank?
      redirect_to :index, alert: "No se ha encontrado el titular de agendas solicitado"
    else
      index
      render :index
    end
  end

```

With this solution you will not need to use rescue block! 

What do you think?

**Comment by raul-fuentes** [comment]
I just removed the rescue block :)

File: app/controllers/visitors_controller.rb [hunk 1]
```diff
@@ -6,13 +6,15 @@ def index
     get_events
     @tree = ancestry_options(Area.unscoped.arrange(:order => 'title')) {|i| "#{'-' * i.depth} #{i.title}" }
     @holders = get_holders_by_area(params[:area])
+    get_holder
```
Comment (Senen) [inline app/controllers/visitors_controller.rb]: @raul-fuentes ¿Why do we need this code here? Maybe i am missing something 🤔 

In this action, holders are chosen from search form selector, so they always exist ...

captura de pantalla 2018-01-10 a las 10 03 08

I think we need redirection only at visitors#agenda action when given param holder does not exist in database

File: app/controllers/visitors_controller.rb [hunk 2]
```diff
@@ -23,6 +25,13 @@ def update_holders
 
   private
 
+  def get_holder
```
Comment (Senen) [inline app/controllers/visitors_controller.rb]: @raul-fuentes if we can remove get_holder call from index action maybe we can remove this method. This code is a little bit difficult to understand

**Comment by Senen** [comment]
Refactored via #322

**Pull request closed without merging**